### PR TITLE
Fixes Issue #875

### PR DESCRIPTION
**Pull request by vedantrathore**
@slint I think this was causing the bug, I have fixed it but still there can be more improvements in the UI, I'll add more commits and squash them later

### CODE REVIEW

**Comment by krzysztof** [comment]
Hey, the fix to #875 is a bit more complex.

If you look at the "Published in" on the right here: https://sandbox.zenodo.org/record/59917/export/json#.WLcmCswrKpF
It just says ", pp. 77", since the metadata is missing the title (look at the JSON  metadata file on the left where it says "part_of" - there's only "pages". If you now look at the whole definition of record metadata here: 

https://github.com/zenodo/zenodo/blob/master/zenodo/modules/records/jsonschemas/records/record-v1.0.0.json#L632-L645

the "part_of" is both the "title" and "pages".

https://github.com/zenodo/zenodo/blob/master/zenodo/modules/records/jsonschemas/records/record-v1.0.0.json#L223-L237

So finally, the task here is to define some jinja logic where the "Published in" is rendered only if there resulting "Published in" label makes sense. It doesn't if the "title" is missing, (hence the "Published in" section shouldn't be rendered at all if there's no title). I guess the same for "Imprint" and "Published in" under journal section below.

**Comment by vedantrathore** [comment]
@krzysztof Do you think we could add some kind of javascript validator function instead of handling it in Jinja?

**Comment by krzysztof** [comment]
@vedantrathore I don't think there's any reason for handling this on the client side. I'd say either Jinja or the view, but Jinja is probably easier.

**Comment by vedantrathore** [comment]
@krzysztof can you tell me where is the relevant section of code for Jinja logic? I'm having trouble finding it

**Comment by krzysztof** [comment]
It in the file/line you have modified, It's only a different conditions that are required in this template, which check for "title" first.

**Comment by vedantrathore** [comment]
@krzysztof Is this is what you meant?

File: zenodo/modules/records/templates/zenodo_records/box/info.html [hunk 1]
```diff
@@ -96,13 +96,33 @@
   {%- if i and not p %}
     <dt>{{_('Imprint')}}:</dt>
     <dd>
-      {{i.publisher}}{% if i.place %}, {{i.place}}{%- endif %}{% if i.isbn %} ({{i.isbn}}){%- endif %}.
+      {{i.publisher}}
```
Comment (krzysztof) [inline zenodo/modules/records/templates/zenodo_records/box/info.html]: Keep this as a one-liner as before

File: zenodo/modules/records/templates/zenodo_records/box/info.html [hunk 2]
```diff
@@ -96,13 +96,33 @@
...
     </dd>
   {%- elif p %}
-    <dt>{{_('Published in')}}:</dt>
-    <dd>
-      {{p.title}}{% if i.place %}, {{i.publisher}}{%- endif %}{% if i.place %}, {{i.place}}{%- endif %}{% if p.pages %}, pp. {{p.pages}}{%- endif %}{% if i.isbn %} ({{i.isbn}}){%- endif %}.
-    </dd>
+      {% if p.title %}
+          <dt>{{_('Published in')}}:</dt>
+          <dd>
+              {{p.title}}
```
Comment (krzysztof) [inline zenodo/modules/records/templates/zenodo_records/box/info.html]: Also a one-liner.

File: zenodo/modules/records/templates/zenodo_records/box/info.html [hunk 3]
```diff
@@ -96,13 +96,33 @@
...
+      {% if i.isbn %}
+          ({{i.isbn}})
+      {%- endif %}.
     </dd>
   {%- elif p %}
-    <dt>{{_('Published in')}}:</dt>
-    <dd>
-      {{p.title}}{% if i.place %}, {{i.publisher}}{%- endif %}{% if i.place %}, {{i.place}}{%- endif %}{% if p.pages %}, pp. {{p.pages}}{%- endif %}{% if i.isbn %} ({{i.isbn}}){%- endif %}.
-    </dd>
+      {% if p.title %}
```
Comment (krzysztof) [inline zenodo/modules/records/templates/zenodo_records/box/info.html]: Combine those two ifs into one: https://github.com/vedantrathore/zenodo/blob/522714f1854d42754df5aaed46edea6f02a0535c/zenodo/modules/records/templates/zenodo_records/box/info.html#L107-L108

File: zenodo/modules/records/templates/zenodo_records/box/info.html (hunk 2 above)
Comment (krzysztof) [inline zenodo/modules/records/templates/zenodo_records/box/info.html]: Also, for the one-liners I meant only in the ``dd`` element :)

**Comment by vedantrathore** [comment]
@krzysztof sorry for the delay, going through my mid semester exams. I have done the requested changes :)

**Comment by krzysztof** [comment]
Okay, thanks. Looks OK, but I'll take a closer look at it later

**Comment by vedantrathore** [comment]
@krzysztof should I squash my commits?

File: zenodo/modules/records/templates/zenodo_records/box/info.html [hunk 4]
```diff
@@ -96,22 +96,24 @@
...
-      {{i.publisher}}{% if i.place %}, {{i.place}}{%- endif %}{% if i.isbn %} ({{i.isbn}}){%- endif %}.
-    </dd>
-  {%- elif p %}
-    <dt>{{_('Published in')}}:</dt>
-    <dd>
-      {{p.title}}{% if i.place %}, {{i.publisher}}{%- endif %}{% if i.place %}, {{i.place}}{%- endif %}{% if p.pages %}, pp. {{p.pages}}{%- endif %}{% if i.isbn %} ({{i.isbn}}){%- endif %}.
+      {{i.publisher}}{% if i.place %}, {{i.place}}{%- endif %}{% if i.isbn %}({{i.isbn}}){%- endif %}.
     </dd>
+  {%- elif p.title %}
+          <dt>{{_('Published in')}}:</dt>
```
Comment (krzysztof) [inline zenodo/modules/records/templates/zenodo_records/box/info.html]: Fix the indentation.

File: zenodo/modules/records/templates/zenodo_records/box/info.html [hunk 5]
```diff
@@ -96,22 +96,24 @@
...
     <dd>
-      {{i.publisher}}{% if i.place %}, {{i.place}}{%- endif %}{% if i.isbn %} ({{i.isbn}}){%- endif %}.
-    </dd>
-  {%- elif p %}
-    <dt>{{_('Published in')}}:</dt>
-    <dd>
-      {{p.title}}{% if i.place %}, {{i.publisher}}{%- endif %}{% if i.place %}, {{i.place}}{%- endif %}{% if p.pages %}, pp. {{p.pages}}{%- endif %}{% if i.isbn %} ({{i.isbn}}){%- endif %}.
+      {{i.publisher}}{% if i.place %}, {{i.place}}{%- endif %}{% if i.isbn %}({{i.isbn}}){%- endif %}.
     </dd>
+  {%- elif p.title %}
```
Comment (krzysztof) [inline zenodo/modules/records/templates/zenodo_records/box/info.html]: Merging of the two conditions from here: https://github.com/zenodo/zenodo/pull/981#discussion_r104384130

should result in ``{%- elif p and p.title %}``. This part will raise an exception if ``p`` is None.

**Comment by krzysztof** [comment]
@vedantrathore Sorry for the delay in reviewing..

**Comment by vedantrathore** [comment]
@krzysztof no problem :), could you review this :).

File: zenodo/modules/records/templates/zenodo_records/box/info.html [hunk 6]
```diff
@@ -96,22 +96,24 @@
...
     <dd>
-      {{i.publisher}}{% if i.place %}, {{i.place}}{%- endif %}{% if i.isbn %} ({{i.isbn}}){%- endif %}.
+      {{i.publisher}}{% if i.place %}, {{i.place}}{%- endif %}{% if i.isbn %}({{i.isbn}}){%- endif %}.
     </dd>
-  {%- elif p %}
+  {%- elif p and p.title %}
     <dt>{{_('Published in')}}:</dt>
     <dd>
-      {{p.title}}{% if i.place %}, {{i.publisher}}{%- endif %}{% if i.place %}, {{i.place}}{%- endif %}{% if p.pages %}, pp. {{p.pages}}{%- endif %}{% if i.isbn %} ({{i.isbn}}){%- endif %}.
+        {{p.title}}{% if i.publisher %}, {{i.publisher}}{%- endif %}{% if i.place %}, {{i.place}}{%- endif %}{% if p.pages %}, pp. {{p.pages}}{%- endif %}{% if i.isbn %}({{i.isbn}}){%- endif %}.
```
Comment (krzysztof) [inline zenodo/modules/records/templates/zenodo_records/box/info.html]: Hey, looks good, one last thing - 
You removed a space after before the (ISBN): ``{% if i.isbn %} ({{i.isbn}}){%- endif %}``, please bring it back and we can merge it :)

File: zenodo/modules/records/templates/zenodo_records/box/info.html [hunk 7]
```diff
@@ -96,22 +96,24 @@
   {%- if i and not p %}
     <dt>{{_('Imprint')}}:</dt>
     <dd>
-      {{i.publisher}}{% if i.place %}, {{i.place}}{%- endif %}{% if i.isbn %} ({{i.isbn}}){%- endif %}.
+      {{i.publisher}}{% if i.place %}, {{i.place}}{%- endif %}{% if i.isbn %}({{i.isbn}}){%- endif %}.
```
Comment (krzysztof) [inline zenodo/modules/records/templates/zenodo_records/box/info.html]: Add space before ``({{i.isbn}})``

**Comment by vedantrathore** [comment]
@krzysztof done :)

**Comment by vedantrathore** [comment]
@krzysztof I can't figure out why the build is failing? It says connection timeout

**Comment by krzysztof** [comment]
@vedantrathore It's a known issue https://github.com/zenodo/zenodo/wiki/Troubleshooting#a-build-in-travis-grid-fails-with-the-following-error
Restarted the build

**Comment by vedantrathore** [comment]
Oh cool :smile:

**Pull request merged**